### PR TITLE
Add handling of plugin configuration

### LINKED ISSUE #25: Separate internal and external go server urls
Add settings `server-host-internal` and `server-host-external`.

Internal one should be used to fetch build information, and external one will be used in the Slack message.

Default both to `127.0.0.1`. Set one to the other, if only one of them is set.

### PR DESCRIPTION
**Pull request by vladistan**
Go version 16.1.0 expects plugins to follow configuration protocol,  which is described here https://developer.go.cd/current/writing_go_plugins/plugin_settings/version_1_0/plugin_settings_configuration.html

In current version the plugins that don't follow this protocol generate errors like the one below.  Most likely in the future they will stop to function.  This pull request implements minimal requirements for config protocol.  This could could be used as jump start to fix #25 and #11

```
Java.lang.RuntimeException: Interaction with plugin with id 'yum' implementing 'package-repository' extension failed while requesting for 'go.plugin-settings.get-configuration'. Reason: [The plugin sent a response that could not be understood by Go. Plugin returned with code '400' and the following response: 'Invalid request name go.plugin-settings.get-configuration']
    at com.thoughtworks.go.plugin.access.PluginRequestHelper.submitRequest(PluginRequestHelper.java:38)
    at com.thoughtworks.go.plugin.access.common.settings.AbstractExtension.getPluginSettingsConfiguration(AbstractExtension.java:45)
    at com.thoughtworks.go.plugin.access.packagematerial.PackageAsRepositoryExtension.getPluginSettingsConfiguration(PackageAsRepositoryExtension.java:51)
    at com.thoughtworks.go.plugin.access.common.settings.PluginSettingsMetadataLoader.fetchPluginSettingsMetaData(PluginSettingsMetadataLoader.java:63)
    at com.thoughtworks.go.plugin.access.common.settings.PluginSettingsMetadataLoader.pluginLoaded(PluginSettingsMetadataLoader.java:47)
    at com.thoughtworks.go.plugin.infra.DefaultPluginManager$FilterChangeListener.pluginLoaded(DefaultPluginManager.java:262)
    at com.thoughtworks.go.plugin.infra.FelixGoPluginOSGiFramework$2.execute(FelixGoPluginOSGiFramework.java:353)
    at org.apache.commons.collections.CollectionUtils.forAllDo(CollectionUtils.java:389)
    at com.thoughtworks.go.plugin.infra.FelixGoPluginOSGiFramework.getBundle(FelixGoPluginOSGiFramework.java:111)
    at com.thoughtworks.go.plugin.infra.FelixGoPluginOSGiFramework.loadPlugin(FelixGoPluginOSGiFramework.java:97)
    at com.thoughtworks.go.plugin.infra.listeners.DefaultPluginJarChangeListener.refreshBundle(DefaultPluginJarChangeListener.java:152)
```

### CODE REVIEW

File: src/main/java/in/ashwanthkumar/gocd/slack/GoNotificationPlugin.java [hunk 1]
```diff
@@ -18,6 +18,7 @@
 
 @Extension
 public class GoNotificationPlugin implements GoPlugin {
+    public static final String CONFIG_VIEW_TEMPLATE = "<div class=\"form_item_block\"><label>Message:</label></div>";
```
Comment (ashwanthkumar) [inline src/main/java/in/ashwanthkumar/gocd/slack/GoNotificationPlugin.java]: Can you please extract this into a separate file ? And also collect 1 property like "Go External URL" from the form? :)

**Comment by ashwanthkumar** [comment]
Thank you so much @vladistan :+1: , for starting work on this. If you could just implement collecting 1 property for the plugin, I guess it would be really easy to add the rest of them. Thoughts ?

File: src/main/java/in/ashwanthkumar/gocd/slack/GoNotificationPlugin.java (hunk 1 above)
Comment (vladistan) [inline src/main/java/in/ashwanthkumar/gocd/slack/GoNotificationPlugin.java]: You mean external with constants?

Comment (ashwanthkumar) [inline src/main/java/in/ashwanthkumar/gocd/slack/GoNotificationPlugin.java]: Extract the HTML into a separate file in the resources. 

Ref
- https://github.com/ind9/gocd-s3-artifacts/blob/master/publish/src/main/resources/views/task.template.html
- https://github.com/ind9/gocd-s3-artifacts/blob/aabcdd7be5fa7fbba0d93dbb6574769d46edbaf5/publish/src/main/java/com/indix/gocd/s3publish/PublishTask.java#L54

Comment (vladistan) [inline src/main/java/in/ashwanthkumar/gocd/slack/GoNotificationPlugin.java]: Ahh.. sure.

File: src/main/java/in/ashwanthkumar/gocd/slack/GoNotificationPlugin.java [hunk 2]
```diff
@@ -47,18 +50,44 @@ public void initializeGoApplicationAccessor(GoApplicationAccessor goApplicationA
...
+        Map<String, Object> response = new HashMap<String, Object>();
+
+        response.put("template", CONFIG_VIEW_TEMPLATE);
+
+        return renderJSON(SUCCESS_RESPONSE_CODE, response);
+    }
+
+    private GoPluginApiResponse handleRequestGetConfiguration() {
+        Map<String, Object> response = new HashMap<String, Object>();
+        Map<String, Object> server_url_params = new HashMap<String, Object>();
```
Comment (ashwanthkumar) [inline src/main/java/in/ashwanthkumar/gocd/slack/GoNotificationPlugin.java]: Nit pick - Can we please use Java's camel case convention?

**Comment by vladistan** [comment]
OK,  implemented the fixes suggested by you

File: src/main/java/in/ashwanthkumar/gocd/slack/GoNotificationPlugin.java (hunk 1 above)
Comment (ashwanthkumar) [inline src/main/java/in/ashwanthkumar/gocd/slack/GoNotificationPlugin.java]: In that case we don't need `CONFIG_VIEW_TEMPLATE` variable right?

**Comment by vladistan** [comment]
Now we good?

**Comment by ashwanthkumar** [comment]
@vladistan I'm merging this now, thanks again :+1:

**Pull request merged**